### PR TITLE
Type conversion for AWS::OpenSearch::Domain

### PR DESCRIPTION
**Pull request by pinzon**
## Motivation
The example from the resource type cdocumentation cannot be deployed in LocalStack due to the lack of some extra type conversions for some parameters.

## Changes
Extra type validations and transformations in the resource provider

## Testing
The added test uses the template from the documentation

### CODE REVIEW

**Review by alexrashed — APPROVED** [review]
Thanks a lot for jumping on this! 🦸🏽 💯 
The code is looking fine, but I am missing a bit of context. Could you please link the documentation you are referring to in your PR description? And what was the issue when deploying the cloudformation template? Could you maybe add a description or a bit of logs?

File: tests/aws/services/cloudformation/resources/test_opensearch.py [hunk 1]
```diff
@@ -48,3 +48,18 @@ def test_domain(deploy_cfn_template, aws_client, snapshot):
...
+    Test that the alternative types for the OpenSearch domain are accepted using the resource documentation example
+    """
+    stack = deploy_cfn_template(
+        template_path=os.path.join(
+            os.path.dirname(__file__), "../../../templates/opensearch_domain_alternative_types.yml"
+        )
+    )
+    domain_name = stack.outputs["SearchDomain"]
+    domain = aws_client.opensearch.describe_domain(DomainName=domain_name)
+    assert domain["DomainStatus"]["DomainName"] == domain_name
```
Comment (dominikschubert) [inline tests/aws/services/cloudformation/resources/test_opensearch.py:65]: This should probably also test the added access policies on the API-level.
Please add snapshots here. :pray:

File: tests/aws/templates/opensearch_domain_alternative_types.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+Resources:
```
Comment (dominikschubert) [inline tests/aws/templates/opensearch_domain_alternative_types.yml:1]:
```suggestion
# sample from https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-opensearchservice-domain.html
Resources:
```

File: localstack/services/opensearch/resource_providers/aws_opensearchservice_domain.py [hunk 3]
```diff
@@ -223,6 +224,19 @@ def create(
...
 
+                for key in ["DedicatedMasterCount", "InstanceCount", "WarmCount"]:
+                    if key in cluster_config and isinstance(cluster_config[key], str):
+                        cluster_config[key] = int(cluster_config[key])
+
+            if properties.get("AccessPolicies") and isinstance(properties["AccessPolicies"], dict):
+                properties["AccessPolicies"] = json.dumps(properties["AccessPolicies"])
+
+            ebs_options = properties.get("EBSOptions")
+            if isinstance(ebs_options, dict):
```
Comment (dominikschubert) [inline localstack/services/opensearch/resource_providers/aws_opensearchservice_domain.py:235]: when should this ever not be a dict?

File: localstack/services/opensearch/resource_providers/aws_opensearchservice_domain.py [hunk 4]
```diff
@@ -223,6 +224,19 @@ def create(
                 cluster_config.setdefault("DedicatedMasterType", "m3.medium.search")
                 cluster_config.setdefault("WarmType", "ultrawarm1.medium.search")
 
+                for key in ["DedicatedMasterCount", "InstanceCount", "WarmCount"]:
+                    if key in cluster_config and isinstance(cluster_config[key], str):
+                        cluster_config[key] = int(cluster_config[key])
+
+            if properties.get("AccessPolicies") and isinstance(properties["AccessPolicies"], dict):
```
Comment (dominikschubert) [inline localstack/services/opensearch/resource_providers/aws_opensearchservice_domain.py:231]: can this actually already be a string? don't think we have a test for this right?

**Review by dominikschubert — CHANGES_REQUESTED** [review]

**Comment by pinzon** [comment]
@alexrashed This is the [documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-opensearchservice-domain.html), it has an example template that cannot be deployed on LocalStack.

```logs
Error applying changes for CloudFormation stack "SampleTestStack2344": Parameter validation failed:
localstack_main  | Invalid type for parameter ClusterConfig.InstanceCount, value: 2, type: <class 'str'>, valid types: <class 'int'>
localstack_main  | Invalid type for parameter ClusterConfig.DedicatedMasterCount, value: 3, type: <class 'str'>, valid types: <class 'int'>
localstack_main  | Invalid type for parameter EBSOptions.Iops, value: 0, type: <class 'str'>, valid types: <class 'int'>
localstack_main  | Invalid type for parameter EBSOptions.VolumeSize, value: 20, type: <class 'str'>, valid types: <class 'int'>
localstack_main  | Invalid type for parameter AccessPolicies, value: {'Version': '2012-10-17', 'Statement': [{'Effect': 'Allow', 'Principal': {'AWS': 'arn:aws:iam::123456789012:user/opensearch-user'}, 'Action': 'es:*', 'Resource': 'arn:aws:es:us-east-1:846973539254:domain/test/*'}]}, type: <class 'dict'>, valid types: <class 'str'> 
```

**Pull request merged**